File: Count Elements With Maximum Frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> counter;
        int maxx = 0;
        int max_count = 0;
        for (auto & num: nums) {
            int val;
            if (counter.find(num) == counter.end()) {
                counter.insert(make_pair(num, 1));
                val = 1;
            } else {
                val = ++counter.at(num);
            }

            if (val > maxx) {
                maxx = val;
                max_count = 1;
            } else if (val == maxx) {
                max_count++;
            }
        }
        return maxx * max_count;
    }
};

File: Count Elements With Maximum Frequency.py
class Solution:
    def maxFrequencyElements(self, nums: List[int]) -> int:
        counter = {}
        max_count = 0
        maxx = 0
        for num in nums:
            if num not in counter:
                counter[num] = 0
            counter[num] += 1

            if counter[num] > maxx:
                maxx = counter[num]
                max_count = 1
            elif counter[num] == maxx:
                max_count += 1
        return max_count * maxx
